src/appleseed.cli/progresstilecallback.cpp

```diff
@@ -90,12 +102,28 @@ namespace
 
             // Print a progress message.
             LOG_INFO(m_logger, "rendering, %s done", pretty_percent(m_rendered_pixels, total_pixels).c_str());
+
+            // Keep track of the total number of rendered tiles.
+            m_rendered_tiles++;
+
+            // Retrieve the total number of tiles in the frame.
+            const size_t total_tiles = frame->image().properties().m_tile_count;
+
+            // Estimate remaining render time.
+            m_stopwatch.measure();
+            const double elapsed_time = m_stopwatch.get_seconds();
+            const double estimated_time = (elapsed_time / m_rendered_tiles) * (total_tiles - m_rendered_tiles);
+
+            // Print estimated time.
+            LOG_INFO(m_logger, "remaining time %.3lf seconds", estimated_time);
```

Comment:
`%.3f` would have been enough here. That said, you should use `pretty_time()` which will print time in a more friendly manner than just seconds. Finally, I don't think we need subsecond precision for time estimations (they are very approximate anyway).

I would also prefer that both the progress message and the remaining time are printed on the same line, something like

    rendering, 53% done; about 5 hours, 30 minutes and 21 seconds remaining...

`pretty_time()` will take care of the latter part.

Btw, make sure to call `c_str()` on the string returned by `pretty_time()`.

Reply:
done :)